src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/IteratorAndAsyncCaptureWalker.cs

```diff
@@ -91,11 +90,10 @@ public static OrderedSet<Symbol> Analyze(CSharpCompilation compilation, MethodSy
                 }
             }
 
+            var variablesToHoist = new OrderedSet<Symbol>();
             if (compilation.Options.OptimizationLevel != OptimizationLevel.Release)
```

Comment:
Is it worth inverting this condition to explicitly call out `Debug` instead of running it for anything that's not Release? I think there are only two options right now, but I could foresee us adding something new in the future

Reply:
I think if we were to add new optimization levels, we'd have to go through everything anyway so it doesn't make much of a difference.

Reply:
I agree with both of you.  The purpose of this test is to improve debugging support, not to optimize release code.  However this is an existing `if` condition and I have no reason to change it in the context of this PR.